cas-server-support-saml/src/main/java/org/jasig/cas/support/saml/util/AbstractSamlObjectBuilder.java

```diff
@@ -264,11 +274,11 @@ public static Document constructDocumentFromXml(final String xmlString) {
      * @param pubKey the pub key
      * @return the element
      */
-    private static org.jdom.Element signSamlElement(final org.jdom.Element element, final PrivateKey privKey,
+    private org.jdom.Element signSamlElement(final org.jdom.Element element, final PrivateKey privKey,
                                                     final PublicKey pubKey) {
         try {
             final String providerName = System.getProperty("jsr105Provider",
-                    "org.jcp.xml.dsig.internal.dom.XMLDSigRI");
+                    SIGNATURE_FACTORY_PROVIDER_CLASS);
```

Comment:
This looks sketchy to me. Nothing about the change you made but the approach generally where we are assuming an internal JDOM provider. There must be a better way to do this. I'll try to do some research and file a separate issue if there seems like a feasible alternative. Again, my concerns are not with changes you made but the approach generally.